docs/rfcs/rfc-006-subscriptions.md

```diff
@@ -0,0 +1,175 @@
+
+```graphql
+subscription {
+  subscribeToMovies(events: [CREATE]) {
+    movie {
+      title
+    }
+  }
+}
+```
+
+If a user wants to get the updates of a particular movie, they could use a `where` argument:
+
+```graphql
+subscription {
+  subscribeToMovies(events: [UPDATE], where: { title: "Titanic" }) {
+    movie {
+      title
+    }
+  }
+}
+```
+
+For subscribing to multiple events, it would be sensible to query also for the `event` field which could then be used for filtering which event triggered the notification:
+
+```graphql
+subscription {
+  subscribeToMovies(events: [CREATE, UPDATE]) {
+    event
+    movie {
+      title
+    }
+  }
+}
+```
+
+If the `event` argument is not provided, it will be assumed that all events want to be listened for:
+
+```graphql
+subscription {
+  subscribeToMovies {
+    event
+    movie {
+      title
+    }
+  }
+}
+```
+
+### Solution 2: Subscription field per event and node type
+
+The subscription type generated by this proposed solution would look like:
+
+```graphql
+type MovieEvent {
+  movie: Movie!
+}
+
+type Subscription {
+  movieCreated(where: MovieWhere): MovieEvent!
+  movieUpdated(where: MovieWhere): MovieEvent!
+  movieDeleted(where: MovieWhere): MovieEvent!
+}
+```
+
+### Usage Examples
+
+If a user wanted to subscribe to all movies being created, they could run the following subscription:
+
+```graphql
+subscription {
+  movieCreated {
+    movie {
+      title
+    }
+  }
+}
+```
+
+If a user wants to get the updates of a particular movie, they could use a `where` argument:
+
+```graphql
+subscription {
+  movieUpdated(where: { title: "Titanic" }) {
+    movie {
+      title
+    }
+  }
+}
+```
+
+For subscribing to multiple events, two subscriptions will be required:
+
+```graphql
+subscription {
+  movieCreated {
+    movie {
+      title
+    }
+  }
+}
+```
+
+```graphql
+subscription {
+  movieUpdated {
+    movie {
+      title
+    }
+  }
+}
+```
+
+## Technical Considerations
+
+* Timestamps as part of event payloads?
+* Single or multiple entities per event payload? (single preferred at present)
+* Do we allow for filtering on properties of related nodes?
+* Do we allow for filtering on relationship properties?
+* Do we allow for projecting relationships in the selection set?
+
+## Risks
+
+* Maintaining order of events being fired
+* Ensure consistency of events data with data in the database
+* Make sure it works across popular PubSub Engine implementations (for example https://www.apollographql.com/docs/apollo-server/data/subscriptions/#production-pubsub-libraries)
+* Make sure it works with `@auth` directive - users shouldn't be able to listen to events for types they can't access
+* Efficiency of Cypher queries - do we fetch all properties of a node and allow GraphQL runtime to filter down, or only the properties in the selection set?
+
+## Out of Scope
+
+* Subscription events for relationship connection and disconnection
```

Comment:
I feel like this _should_ be in scope as relationships between nodes are _at least_ half of what makes Neo4j special and it's important to track those events.